libs/datamodel/src/dml/validator/directive/core/unique.rs

```diff
@@ -61,7 +61,14 @@ impl DirectiveValidator<dml::Model> for ModelLevelUniqueValidator {
             .fields
             .iter()
             .filter_map(|field| {
-                if obj.find_field(&field).is_none() {
+                if obj
+                    .fields
+                    .iter()
+                    .find(|model_field| {
+                        model_field.database_name.as_ref().unwrap_or(&model_field.name) == field.as_str()
+                    })
+                    .is_none()
```

Comment:
this is backwards (the field names should be the model field names, not the database field names)